FWCore/MessageLogger/src/JobReport.cc

```diff
@@ -725,7 +742,7 @@ namespace edm {
   JobReport::dumpFiles(void) {
     std::ostringstream msg;
 
-
+    tinyxml2::XMLDocument doc;
```

Comment:
Will having the `doc` outside of the loop mean it will accumulate memory until the loop finishes?

Reply:
The last commit is using a method to delete the document nodes when the doc is used to create new text objects inside loops.